developing/vcs/README.md

```diff
@@ -2,7 +2,321 @@
+hasn't been reviewed yet, run the following commands:
+
+```
+git checkout old_branch
+git branch -m new_branch                   # Rename branch locally
+git push origin --delete old_branch        # Delete the old remote branch
+git push --set-upstream origin new_branch  # Push the new branch, set local branch to track the new remote
+```
+
+As you work locally, before pushing to GitHub, feel free to amend and/or squash
+your commits to keep the history meaningful and concise. For example, if you
+made a commit, and then realized you made a typo, you can amend the previous
+commit:
+
+```
+git commit --amend
+```
+
+When the feature/fix is complete and tests pass, rebase to incorporate upstream
+changes.
+
+```
+git fetch origin
+git rebase origin/master
+```
+
+Resolve any conflicts manually. If the conflicted file was deleted on the master
+branch, remove it from your branch:
+
+```
+git rm <conflicted_file>
+```
+
+Otherwise, add it:
+
+```
+git add <conflicted_file>
+```
+
+Continue the rebase:
+
+```
+git rebase --continue
+```
+
+If you've created more than one commit, use
+`[git's interactive rebase](https://help.github.com/en/articles/about-git-rebase)`
+to squash them into cohesive commits with good messages. Each commit must be
+green (builds and passes tests, CI, and other checks):
+
+```
+git rebase -i origin/master
+```
+
+For each meaningful commit, write a
+[good commit message](https://chris.beams.io/posts/git-commit/) following these
+seven rules:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain what and why vs. how
+
+Example:
+
+```
+Replace chromedriver-helper with webdrivers
+
+`chromedriver-helper` says in bold on their README that they are no
+longer maintaining it and that people should use `webdrivers` instead.
+
+Additionally, `webdrivers` has a very useful feature that will
+automatically use the right version of `chromedriver` based on the
+version of Chrome that is installed on the machine. This was not
+possible with `chromedriver-helper`, which made it harder to deal with
+versioning locally and in Circle CI.
+
+Resolves #123
+```
+
+Push your branch:
+
+```
+git push origin <branch-name>
+```
+
+Submit a GitHub pull request.
+
+Ask for a [code review](../code-reviews/README.md) in the project's Slack
+channel, and/or via the GitHub site.
+
+Once you push to GitHub, only make changes via new commits, as opposed to
+amending or rebasing, to make it easier for reviewers to see what has changed
+since they last provided feedback. If you need to make a minor change to an
+earlier commit, take advantage of the `--fixup` flag, coupled with the
+`--autosquash` flag for interactive rebase, which you should enable globally:
+
+```
+git config --global rebase.autosquash true
```

Comment:
Again: explain why, and explain (if possible, this is git after all) what this is doing.